src/partition/ReceivedPartition.cpp

```diff
@@ -458,19 +443,6 @@ void ReceivedPartition::compareBoundingBoxes()
   }
 }
 
-bool ReceivedPartition::overlapping(mesh::Mesh::BoundingBox const &currentBB, mesh::Mesh::BoundingBox const &receivedBB)
-{
-  // Comparison is done for all dimensions and, of course, to have a proper overlap, each dimension must overlap.
-  // We need to check if first AND second is smaller than first of the other BB to prevent false negatives due to empty bounding boxes.
-  for (size_t i = 0; i < currentBB.size(); i++) {
-    if ((currentBB[i].first < receivedBB[i].first && currentBB[i].second < receivedBB[i].first) ||
-        (receivedBB[i].first < currentBB[i].first && receivedBB[i].second < currentBB[i].first)) {
-      return false;
-    }
-  }
-  return true;
-}
-
 void ReceivedPartition::prepareBoundingBox()
```

Comment:
Maybe we don't need that function anymore. All could just happen inline. Should be easier to understand.